### ISSUE TITLE
Displaced chart with extreme spline tension values

### ISSUE THREAD

**Issue by jwoLondon** [vega/vega-lite#5598]
Not sure where this sits on the _bug <—> feature request_ spectrum, but I've logged it here as bug as it results in unexpected behaviour...

When interpolating a line using cardinal splines with extreme `tension` values outside [0, 1], the plot can be unexpectedly displaced. This is includes the entire plot area, and the data view rectangle relative to the title. The displacement can be beyond the visible path of the interpolated line, making it difficult to compose multiple views.

![splineTension](https://user-images.githubusercontent.com/1846999/69873024-176fc880-12af-11ea-9d2c-beb31b38835c.png)

While the schema suggests the `tension` value should be between 0 and 1, there are definitely cases where values outside this range are useful (e.g. [this example](https://github.com/jwoLondon/30dayMapChallenge/blob/master/d29Experimental.md), but even setting `tension` to -1 is useful for creating rounded shapes:

<img width="370" alt="Screen Shot 2019-11-29 at 13 39 48" src="https://user-images.githubusercontent.com/1846999/69873091-4dad4800-12af-11ea-851d-1d82124c2ffc.png">

Setting `"clip" : true` helpfully clips the view to the rectangle defined by the axes, but it would be more useful if the unclipped area was that of the larger of this rectangle and the minimum enclosing rectangle the interpolated line path. I would also propose that the [0, 1] tension range be removed from the schema given legitimate uses of values beyond this range.

Minimum working example:

```json
{
  "$schema": "https://vega.github.io/schema/vega-lite/v4.json",
  "width": 300,
  "height": 300,
  "title": { "text": "Overshooting splines"},
  "background": "papayawhip",
  "data": {
    "values": [
      { "x": 20, "y": 10},
      { "x": 30, "y": 30},
      { "x": 50, "y": 10}
    ]
  },
  "encoding": {
    "x": { "field": "x", "type": "quantitative"},
    "y": { "field": "y", "type": "quantitative"}
  },
  "mark": {
    "type": "line",
    "tension": -7,
    "clip": false,
    "interpolate": "cardinal-closed",
    "point": { "stroke": "black"}
  }
}
```

**Comment by jheer** [comment]
The displacement you're seeing here stems from the naive bounds calculation that vega-scenegraph uses for bezier curves: [we simply include all control points](https://github.com/vega/vega/blob/master/packages/vega-scenegraph/src/bound/boundContext.js#L33).

If we can identify a more accurate -- but computationally cheap -- method, then the extreme displacement could be avoided and hopefully no clipping is necessary.

**Comment by jwoLondon** [comment]
Thanks Jeff. 

[De Casteljau's algorthm](https://stackoverflow.com/a/13378980) looks promising, although I don't know in practice how 'computationally cheap' you need it to be.

**Comment by jheer** [comment]
I'm looking into it. I think we can directly solve for extrema (set the derivative of the Bezier equations to zero) rather than use an iterative method.

**Comment by domoritz** [comment]
Should we change the schema to allow tension values outside of [0,1]?

**Comment by jheer** [comment]
I tightened the bounds calculations in #2213, will be included in Vega v5.9.0.

I don't see why the schema can't be more permissive.

**Comment by domoritz** [comment]
I'll also fix Vega's typings in https://github.com/vega/vega/blob/b23c9800c46adaec8c1ffc04f542c544c20ad817/packages/vega-typings/types/spec/config.d.ts#L184.
